app/dependency_source_code_finders/node.rb

```diff
@@ -16,12 +16,11 @@ def look_up_github_repo
         all_versions.map { |v| v["homepage"] } +
         all_versions.map { |v| get_url(v.fetch("bugs", {})) }
 
-      potential_source_urls = potential_source_urls.reject(&:nil?)
+      potential_source_urls = potential_source_urls.compact
 
       source_url = potential_source_urls.find { |url| url =~ GITHUB_REGEX }
 
-      @github_repo =
-        source_url.nil? ? nil : source_url.match(GITHUB_REGEX)[:repo]
+      @github_repo = source_url && source_url.match(GITHUB_REGEX)[:repo]
```

Comment:
Not a huge fan of short circuit operators in Ruby...

You could use `source_url&.match(...)&.dig(:repo)` instead.